lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
@adamjstewart @davydden @pramodk: I think the tricky thing here is that as @serbanmaerean says, `xl_r` isn't just extra flags; it's extra runtime libraries that come from a rather esoteric config file bundled with the XL compilers.  The "right" libraries/flags are selected based on the name the compiler is invoked with, and my understanding is that can change from release to release.  Given that I think it's effectively a second compiler.  If IBM one day changes that interface we could consider doing this differently.